EFCore/MAUI/Readme.md

```diff
@@ -0,0 +1,873 @@
+
+   _API/CustomEndPointController.cs_:
+   ```cs
+   [HttpGet(nameof(CanCreate))]
+   public IActionResult CanCreate(string typeName) {
+       var strategy = (SecurityStrategy)_securityProvider.GetSecurity();
+       var objectType = strategy.TypesInfo.PersistentTypes.First(info => info.Name  == typeName).Type;
+       return Ok(strategy.CanCreate(objectType));
+   }
+   ```
+
+2. Add a method that checks the permissions and a method that submits a new post to the .NET MAUI application's data service.
+
+   _Services/IDataStore.cs_:
+
+   ```cs
+   public interface IDataStore<T> {
+       // ...
+       Task<bool> UserCanCreatePostAsync();
+       Task<T> AddItemAsync(Post post);
+   }
+   ```
+
+   _Services/WebAPIService.cs_:
+   
+   ```cs
+   public class WebAPIService : IDataStore<Post> {
+      // ...
+       public async Task<bool> UserCanCreatePostAsync() 
+	        => (bool)JsonNode.Parse(await HttpClient.GetStringAsync($"{_apiUrl}  CustomEndpoint/CanCreate?typename=Post"));
+
+       public async Task<bool> AddItemAsync(Post post) {
+           var httpResponseMessage = await HttpClient.PostAsync(_postEndPointUrl,
+               new StringContent(JsonSerializer.Serialize(post), Encoding.UTF8, ApplicationJson));
+           if (!httpResponseMessage.IsSuccessStatusCode) {
+               await Shell.Current.DisplayAlert("Error", await httpResponseMessage.Content.ReadAsStringAsync(), "OK");
+           }
+           return httpResponseMessage.IsSuccessStatusCode;
+       }
+   }
+   ```
+
+3. Configure the views and navigation:
+
+   _ViewModels/ItemsViewModel.cs_:
+
+   ```cs
+   public class ItemsViewModel : BaseViewModel {
+      public ItemsViewModel() {
+         // ...
+         AddItemCommand = new Command(OnAddItem);
+      }
+      public Command AddItemCommand { get; }
+
+      async void OnAddItem(object obj) {
+         if (await DataStore.UserCanCreatePostAsync()) {
+               await Navigation.NavigateToAsync<NewItemViewModel>(null);
+         }
+         else {
+               await Shell.Current.DisplayAlert("Error", "Access denied", "Ok");
+         }
+      }
+   }
+   ```
+
+   _App.xaml.cs_:
+
+   ```cs
+   public App() {
+      // ...
+      Routing.RegisterRoute(typeof(NewItemPage).FullName, typeof(NewItemPage));
+      // ...
+   }
+   ```
+
+
+   _Views/ItemsPage.xaml_:
+
+   ```xml
+   <ContentPage.ToolbarItems >
+       <ToolbarItem Text="Add" Command="{Binding AddItemCommand}" />
+   </ContentPage.ToolbarItems>
+   ```
+
+![](../../images/MAUI/Edit.png)
+   
+### Allow users to archive records
+
+1. In the Web API service, create the `Archive` endpoint, which gets posts from the database, and archives them to the disk. The controller action's implementation bellow uses the `securedObjectSpaceFactory` to communicate with the data store so that all operation on the data respect the security permissions. 
```

Comment:
```suggestion
1. In the Web API service, create an `Archive` endpoint. This endpoint will obtain a post from the database and archive that post to the disk. The controller action's implementation below uses the `securedObjectSpaceFactory` to communicate with the data store so that all data operations respect security permissions. 
```